### DISCUSSION TITLE
Support of Matroska Audio files (.mka)

### DISCUSSION THREAD

**Discussion by t-richter2** [TortugaPower/BookPlayer#580] [Ideas]
Hi,
I have a huge collection of mp3 audiobooks (all legal, of course).

Rather than reencoding them (which would take literally ages on my computer) I would prefer to just wrap them into a Matroska container.
This could potentially contain all sorts of files with chapters, cover art etc.

Plus, it would be a USP, since no other app on iOS (at least I know of) currently supports it. ;-)

**Comment by pichfl** [comment]
It is unclear if we can support https://github.com/Matroska-Org/libmatroska in our project. As you say, I don't think anyone has tried this on iOS before (except VLC most likely) so the effort involved is considerably large. 

We have to balance new features against the time we can invest into the project. This means it is very unlikely that we will tackle this in the near future as there are more pressing matters which will benefit a large portion of our users compared to supporting a rather obscure new format. 

Good news though: We will provide new ways to organize mp3 based books in the future which do not require encoding at all.

**Reply by t-richter2** [reply]
Ok, I see... I still hope for some upvotes, though ;-)
Matroska is around for about 15 years and is pretty much the standard for video files on the webs.
I think libVLC is also under GPL, so it could - in theory -  be used?
If there's any interest I could have a look at it - but I also understand if you'd rather have no dependency on libVLC.

**Reply by pichfl** [reply]
There is indeed a [VLCKit](https://code.videolan.org/videolan/VLCKit/tree/master/Examples/iOS/DropIn-Player) drop-in library that we need to take a look at. 

The big question is how efficient the encoding actually is. We have some users reporting battery drain issues which are caused by the encoding not being routed through the decoder hardware if iOS encounters problems with the files. So we need to check if VLCKit does support hardware encoding otherwise it will be quite heavy on the battery. 

Thanks for bringing this up 🙂
